clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3080,12 +3084,40 @@ private int processKafkaDataMessage(
         // update checksum for this PUT message if needed.
         partitionConsumptionState.maybeUpdateExpectedChecksum(keyBytes, put);
 
-        // Do transorfmation recompute key, value and partition
+        ByteBuffer valueBytes = put.getPutValue();
+        int putSchemaId = put.getSchemaId();
+        Schema valueSchema = schemaRepository.getValueSchema(storeName, putSchemaId).getSchema();
+
+        // Decompress/assemble record
```

Comment:
We should only be doing any of this if the record transformer is enabled, so we should move this logic into the if block (if (recordTransformer != null)